FlowCrypt/Core/Models/KeyDetails.swift

```diff
@@ -40,6 +40,20 @@ extension KeyDetails {
     var pgpUserEmails: [String] {
         users.map { MCOAddress(nonEncodedRFC822String: $0).mailbox }
     }
+    
+    var isKeyUsuable: Bool {
+        // revoked keys are not usable
+        guard !revoked else { return false }
+        // keys without lastModified don't have valid signatures on them - not usable
+        guard lastModified != nil else { return false }
+        // keys without uids on them are not usable
+        guard users.isNotEmpty else { return false }
+        // expired keys are not usable
+        if let expiration = expiration, expiration.toDate() < Date() { return false }
```

Comment:
You can use `timeIntervalSinceNow` here, so no need to initialise another `Date()`:
```swift
if let expiration = expiration, expiration.timeIntervalSinceNow < 0 { return false }
```